codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -95,6 +95,7 @@ type rec t = {
   //of a rollback.
   dynamicContracts: DynamicContractsMap.t,
   isFetchingAtHead: bool,
+  firstEventBlockNumber: option<int>,
```

Comment:
Solution is to add firstEventBlockNumber to each partition and evaluate the earliest across partitions.